File: packages/camera/camera_platform_interface/lib/src/types/take_picture_result.dart
import 'package:camera_platform_interface/camera_platform_interface.dart';

import 'take_picture_mode.dart';

/// The result of a picture taking
class TakePictureResult {
  /// The result file
  final XFile file;

  /// Indicates if the vertical tilt's calculation is available for the device
  final bool isVerticalTiltAvailable;

  /// Indicates if the horizontal tilt's calculation is available for the device
  final bool isHorizontalTiltAvailable;

  /// The value of the vertical tilt
  final double verticalTilt;

  /// The value of the horizontal tilt
  final double horizontalTilt;

  /// The mode of the picture taking
  final TakePictureMode mode;

  /// Default constructor
  TakePictureResult({
    required this.file,
    required this.isVerticalTiltAvailable,
    required this.isHorizontalTiltAvailable,
    required this.verticalTilt,
    required this.horizontalTilt,
    required this.mode,
  });

  /// Converts the supplied [Map] to an instance of the [TakePictureResult]
  /// class.
  TakePictureResult.fromJson(Map<dynamic, dynamic> json)
      : verticalTilt = json['verticalTilt'],
        horizontalTilt = json['horizontalTilt'],
        isHorizontalTiltAvailable = json['isHorizontalTiltAvailable'],
        isVerticalTiltAvailable = json['isVerticalTiltAvailable'],
        file = XFile(json['resultPath']),
        mode = (json['mode'] as String).toTakePictureMode;

  /// Converts the [TakePictureResult] instance into a [Map] instance that
  /// can be serialized to JSON.
  Map<String, dynamic> toJson() =>
      {
        'verticalTilt': verticalTilt,
        'horizontalTilt': horizontalTilt,
        'isHorizontalTiltAvailable': isHorizontalTiltAvailable,
        'isVerticalTiltAvailable': isVerticalTiltAvailable,
        'resultPath': file.path,
        'mode': mode.toStringValue,
      };

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
          other is TakePictureResult &&
              runtimeType == other.runtimeType &&
              file == other.file &&
              isVerticalTiltAvailable == other.isVerticalTiltAvailable &&
              isHorizontalTiltAvailable == other.isHorizontalTiltAvailable &&
              verticalTilt == other.verticalTilt &&
              horizontalTilt == other.horizontalTilt &&
              mode == other.mode;

  @override
  int get hashCode =>
      file.hashCode ^
      isVerticalTiltAvailable.hashCode ^
      isHorizontalTiltAvailable.hashCode ^
      verticalTilt.hashCode ^
      horizontalTilt.hashCode ^
      mode.hashCode;

  @override
  String toString() {
    return 'TakePictureResult{file: $file, isVerticalTiltAvailable: $isVerticalTiltAvailable, isHorizontalTiltAvailable: $isHorizontalTiltAvailable, verticalTilt: $verticalTilt, horizontalTilt: $horizontalTilt, mode: $mode}';
  }
}

File: packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/types/TakePictureResult.java
package io.flutter.plugins.camera.types;

import java.util.HashMap;

public class TakePictureResult {
    public final String resultPath;
    public final double horizontalTilt;
    public final double verticalTilt;
    public final boolean isHorizontalTiltAvailable;
    public final boolean isVerticalTiltAvailable;
    public final TakePictureMode mode;
    public final int width;
    public final int height;


    public TakePictureResult(String resultPath,
                             double horizontalTilt,
                             double verticalTilt,
                             boolean isHorizontalTiltAvailable,
                             boolean isVerticalTiltAvailable,
                             TakePictureMode mode,
                             int width,
                             int height) {
        this.resultPath = resultPath;
        this.horizontalTilt = horizontalTilt;
        this.verticalTilt = verticalTilt;
        this.isHorizontalTiltAvailable = isHorizontalTiltAvailable;
        this.isVerticalTiltAvailable = isVerticalTiltAvailable;
        this.mode = mode;
        this.width = width;
        this.height = height;
    }


    public HashMap<String, Object> getMap() {
        return new HashMap<String, Object>() {
            {
                put("resultPath", resultPath);
                put("isHorizontalTiltAvailable", isHorizontalTiltAvailable);
                put("isVerticalTiltAvailable", isVerticalTiltAvailable);
                put("horizontalTilt", horizontalTilt);
                put("verticalTilt", verticalTilt);
                put("mode", mode.toString());
                put("width", width);
                put("height", height);
            }
        };
    }
}